aodndata/nsw_oeh/handler.py

```diff
@@ -0,0 +1,434 @@
+
+        # Overall checks...
+        messages = []
+
+        # metadata sheet (PDF) exists
+        if not have_metadata:
+            messages.append("Missing metadata file (PDF format)")
+
+        # shapefile exists
+        # TODO: Check there is exactly one coverage shapefile
+        if not have_coverage:
+            messages.append("Missing survey coverage shapefile")
+
+        # at least one XYZ file for MB surveys
+        if not have_xyz and self.survey_methods == 'MB':
+            messages.append("Missing bathymetry xyz file")
+
+        if messages:
+            report["Zip file contents"] = messages
+
+        return report
+
+    def get_dest_path(self):
+        """
+        Return the relative path where the contents of the zip file should be
+        published to on S3. Raise an exception if required metadata are missing.
+        :return: relative path
+        :rtype: str
+        """
+        survey_year = self.survey_name[:4]
+        methods_name = SURVEY_METHODS.get(self.survey_methods, '')
+        if not methods_name or not survey_year or not self.survey_name:
+            raise ValueError(
+                "Could not determine destination path for {zip}".format(zip=self.zip_file)
+            )
+        return os.path.join('NSW-OEH', methods_name, survey_year, self.survey_name)
+
+    def extract(self, tmp_dir):
+        """
+        Extract files from the zip file into tmp_dir in preparation for publishing. Directory
+        structure within the zip file is ignored. For a multi-beam survey, all files are
+        extracted. For "basic packaged" files (single-beam or lidar/lads), only the coverage
+        shapefile is extracted, and the zip file itself is copied into tmp_dir.
+        :param tmp_dir: Full path of temporary directory to extract into
+        :return: Names of files to be published (within tmp_dir)
+        :rtype: list
+        """
+        publish_files = []
+        with zipfile.ZipFile(self.zip_file) as zf:
+            for zip_name in self.zip_contents:
+                file_name = os.path.basename(zip_name)
+
+                if self.survey_methods == 'MB' or SHAPEFILE_PATTERN.match(file_name):
+                    ext_path = zf.extract(zip_name, tmp_dir)
+
+                    # Move file directly into base of tmp_dir, out of any directories in the zip file
+                    if zip_name != file_name:
+                        shutil.move(ext_path, tmp_dir)
+
+                    publish_files.append(file_name)
+
+        if self.survey_methods in BASIC_PACKAGED_METHODS:
+            shutil.copy(self.zip_file, tmp_dir)
+            publish_files.append(os.path.basename(self.zip_file))
+
+        return publish_files
+
+
+class NswOehHandler(HandlerBase):
+    def __init__(self, *args, **kwargs):
+        super(NswOehHandler, self).__init__(*args, **kwargs)
+
+        self.allowed_extensions = ['.zip']
+        self.campaign_folder_path = None
```

Comment:
Minor point, but `survey_path` would be more consistent with the terminology used in the rest of the code (and used by NSW OEH folks)

Reply:
done